src/content/resources/javascript/es6/arrow-functions.md

```diff
@@ -1,11 +1,70 @@
 ---
 authors:
   - "veksen#1060"
+  - "t0m#5956"
 created_at: 2019/10/08
-updated_at: 2019/10/08
+updated_at: 2019/10/13
 title: Arrow functions
 ---
+## Arrow Function Syntax
 
-## Arrow functions
+Arrow functions are a new way to declare functions within ES6. Arrow functions allow for a shorter syntax. For example, below is a regular function, as usually seen before ES6.
+```javascript
+const sayHello = function() {
+  console.log('hello');
+}
+```
+The same function can be rewritten with arrow functions, as seen below.
+```javascript
+const sayHello = () => {
+  console.log('hello');
+}
+```
+Both of the above functions have no parameters, but if you want to pass parameters into the function with the arrow syntax, you can do the following, placing the parameters within the brackets of the arrow function.
+```javascript
+const sayHello = (nameOne,nameTwo) => {
+  console.log(nameOne,'says hello to',nameTwo);
+}
+```
+If you want to pass only one parameter, you don't need the brackets around the parameters, and can do the following:
+```javascript
+const sayHello = name => {
+  console.log('hello',name);
+}
+```
+Another interesting feature that also reduces the syntax within arrow functions, is that if a function only returns an expression using one line of code, you can omit the ``return`` keyword and the curly braces. Be aware that in multiple line arrow functions, the return keyword still needs to be used. The following function returns "hello":
+```javascript
+const sayHello = () => "hello";
+```
+## Handling of the Keyword This
+The keyword ``this`` is handled differently in arrow functions - in arrow functions, ``this`` is not bound to anything. This means that the keyword ``this`` represents the object that defined the function in an arrow function, whereas in a standard function it typically represents the object that called the function.
 
-// TODO: finish this
+In this snippet of code, it is a regular function, so the keyword this represents the HTMLButtonObject that called it, and therefore the function outputs "[object HTMLButtonElement]" to the console.
+```js
+testEvent = function() {
+  console.log(this);
+}
+
+//A button object calls the function:
+document.querySelector("#btn").addEventListener("click",testEvent);
+```
+However, in this snippet of code, an arrow function is used, and the function outputs "[object Window]", as that is the object that defined the function.
+```js
+testEvent = () => {
+  console.log(this);
+}
+
+//A button object calls the function:
+document.querySelector("#btn").addEventListener("click",testEvent);
+```
+In some instances, arrow functions are required. This can be seen in the constructor method in a class. This requires the arrow function type syntax, like the following ``constructor(){}``. An example of a constructor method being used in a class can be seen below:
```

Comment:
Sorry if I'm being picky, but I don't see how this sentence makes sense, you're saying sometimes arrow functions are required but there are no arrow functions in the example below

Reply:
Yeah, not so sure what I typed....